### PR TITLE
Add rke2 Security Scan Files

### PR DESCRIPTION
**Pull request by briandowns**
This adds the files necessary for being able to run security-scan against an RKE2 cluster.

If approved, I'll squash merge.

Reference issue: https://github.com/rancher/rke2/issues/1

### CODE REVIEW

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (brandond) [inline pkg/kb-summarizer/summarizer/summarizer.go:375]: What's our convention on commented out code?

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (brandond) [inline pkg/kb-summarizer/summarizer/summarizer.go:481]: Same question

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (briandowns) [reply pkg/kb-summarizer/summarizer/summarizer.go:375]: Not sure. The code isn't used and didn't want to remove it. figured I'd comment it out and the primary maintainer could make the call. I'd say for anything rke2 related, it gets removed entirely but not sure for this repo.

**Comment by briandowns** [comment]
@prachidamle Can I bother you for review?

File: package/Dockerfile
Comment (prachidamle) [inline package/Dockerfile:40]: Indentation seems to be missing one space? It looked correct earlier

File: package/cfg/config.yaml
Comment (prachidamle) [inline package/cfg/config.yaml:202]: We will also need to update the CIS v2 chart with corresponding ClusterScanBenchmark and ClusterScanProfile to use this version

File: package/cfg/rke2-cis-1.5/etcd.yaml
Comment (prachidamle) [inline package/cfg/rke2-cis-1.5/etcd.yaml:21]: Do we need any custom remediations for rke2 or the default will apply?

File: package/cfg/rke2-cis-1.5/policies.yaml
Comment (prachidamle) [inline package/cfg/rke2-cis-1.5/policies.yaml:13]: We have policies audit scripts implemented for rke, they are pretty generic kubectl commands, and I think rke2 can re-use those instead of calling these as  type: "manual"

File: package/run.sh
Comment (prachidamle) [inline package/run.sh:7]: Does listing these here help in the run.sh?

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (prachidamle) [inline pkg/kb-summarizer/summarizer/summarizer.go:503]: Can we keep this method commented, this gives us an easy reference of how we translate kube-bench states to our report.

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (prachidamle) [inline pkg/kb-summarizer/summarizer/summarizer.go:621]: Do we need this at Info?

File: package/Dockerfile
Comment (briandowns) [reply package/Dockerfile:40]: Indentation was 5 and I updated it to the standard 4.

File: package/cfg/rke2-cis-1.5/etcd.yaml
Comment (briandowns) [reply package/cfg/rke2-cis-1.5/etcd.yaml:21]: Defaults should apply.

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (briandowns) [reply pkg/kb-summarizer/summarizer/summarizer.go:503]: Can we include this in soime form of documentation instead? Like a package doc?

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:78]: maybe copied from elsewhere, but why does the test for 1.1.1 have 7 different perms it can test for and this only have one? I see the latest iteration for rke1 has this fixed:
https://github.com/rancher/security-scan/blob/master/package/cfg/rke-cis-1.5-hardened/master.yaml#L74

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:192]: This audit doesnt work. `--data-dir` is not passed as a command line argument to the etcd process. its specified in the etcd config file at `/var/lib/rancher/rke2/server/db/etcd/config`

Also, the rke1 test seems to have been simplified to just a stat: https://github.com/rancher/security-scan/blob/master/package/cfg/rke-cis-1.5-hardened/etcd.yaml#L11-L20

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:206]: Should not be skipped. And generally if something is skipped, the remediation should explain why.

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:222]: should not be skipped

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:210]: Same as 1.1.11. this audit wont work

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:225]: This represents the the admin's kubeconfig file. RKE2 does in fact put such a file on disk, but at a different location (`/etc/rancher/rke2/rke2.yaml`).

This audit should be updated to test that.

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:256]: Cannot skip for same reasons at 1.1.13

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:288]: If we are skipping, we need to provide a reason in the remediation. See how its done in rke1:
/var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:328]: This audit wont work. File is at `/var/lib/rancher/rke2/server/tls`

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:329]: Should not be manual. rke1 actually has a script for it:
https://github.com/rancher/security-scan/blob/master/package/cfg/rke-cis-1.5-hardened/etcd.yaml#L46

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:335]: should not be skipped

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:346]: See comments for 1.1.19. shouldnt be skipped and audit and remediation need updated

File: package/cfg/rke2-cis-1.5/master.yaml
Comment (cjellick) [inline package/cfg/rke2-cis-1.5/master.yaml:357]: See comments for 1.1.19. shouldnt be skipped and audit and remediation need updated

**Review by brandond — APPROVED** [review]

**Comment by prachidamle** [comment]
@briandowns can you squash all commits so that the lines you had added to debug will get cleaned up and gets easier to review?

**Review by brandond — DISMISSED** [review]

**Review by brandond — APPROVED** [review]

File: package/run.sh
Comment (mrajashree) [inline package/run.sh:82]: are we testing these script changes for RKE1 profile also?

Comment (briandowns) [reply package/run.sh:82]: I tested in RKE1. Everything ran as expected.

**Comment by briandowns** [comment]
> @briandowns can you squash all commits please so that the lines you had added to debug will get cleaned up and gets easier to review?

I spent a few hours trying to squash this with help from a couple folks and it appears as though it's going to be too time consuming and complicated to be worth it. I can still squash this upon merge though so we'll only add 1 commit for the work.

**Comment by davidnuzik** [comment]
Can we get eyes on this? I'd like to move this along. If we need to wait until tomorrow or something due to Rancher releases today that's fine just don't want it to sit. We can squash when merge.

File: package/cfg/config.yaml
Comment (prachidamle) [inline package/cfg/config.yaml:185]: are we sure we need this? Maybe can we check by removing this change to see if it still runs the tests under policies, seems like we did not need this for other profiles having policies.

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (prachidamle) [inline pkg/kb-summarizer/summarizer/summarizer.go:718]: We need not print all these individually and just print the object as done by this previous commit: https://github.com/rancher/security-scan/pull/25/files

File: pkg/kb-summarizer/summarizer/summarizer.go
Comment (prachidamle) [inline pkg/kb-summarizer/summarizer/summarizer.go:733]: Can we remove the individual print lines

**Comment by prachidamle** [comment]
Have some comments, but rest of the changes are lgtm

**Comment by briandowns** [comment]
> Have some comments, but rest of the changes are lgtm

@prachidamle 

Updates made, tested, and pushed. Thanks!

**Review by prachidamle — APPROVED** [review]
LGTM, pls squash to single commit when merging

**Review by galal-hussein — APPROVED** [review]
LGTM

**Review by mrajashree — APPROVED** [review]
Still relatively new to understanding CIS/kube-bench so I don't have enough context w.r.t script changes, but looks good

**Pull request merged**
